webui/react/src/pages/TrialDetails/TrialInfoBox.tsx

```diff
@@ -44,10 +66,86 @@ const TrialInfoBox: React.FC<Props> = ({ trial, experiment }: Props) => {
   const [canceler] = useState(new AbortController());
   const [models, setModels] = useState<Loadable<ModelItem[]>>(NotLoaded);
   const [selectedModelName, setSelectedModelName] = useState<string>();
+  const shouldRenderAllocationCard = useMemo(
+    () => trial !== undefined && experiment.numTrials === 1,
+    [trial, experiment],
+  ); // as per ticket requirements, we're only rendering it on single trial experiments and trial details pages
+  const resourcePools = Loadable.getOrElse([], useObservable(clusterStore.resourcePools));
+
+  const taskAllocation = useAsync(async () => {
+    if (!trial) return;
+
+    // one big issue is that taskIds is an optional property
+    const taskId = trial.taskIds !== undefined ? trial.taskIds[trial.taskIds.length - 1] : '';
```

Comment:
`trial.taskIds.length - 1` can be-1 if `trial.taskIds.length` is 0. 

> one big issue is that taskIds is an optional property

can you change it to a non-optional property